cmd/nvidia-dra-controller/imex.go

```diff
@@ -312,3 +310,59 @@ func (m *ImexManager) cleanupResourceSlices() error {
 
 	return nil
 }
+
+// add sets the offset where an IMEX domain's channels should start counting from.
+func (offsets imexDomainOffsets) add(imexDomain string, resourceSliceImexChannelLimit, driverImexChannelLimit int) (int, error) {
+	id := strings.SplitN(imexDomain, ".", 2)
+	if len(id) != 2 {
+		return -1, fmt.Errorf("error adding IMEX domain %s: invalid format", imexDomain)
+	}
+	domain := id[0]
+	cliqueID := id[1]
+
+	// Check if the IMEX domain is already in the map.
+	if _, ok := offsets[domain]; !ok {
+		offsets[id[0]] = make(map[string]int)
+	}
+
+	// Check if the clique is already in the map.
+	var offset int
+	if _, ok := offsets[domain][cliqueID]; !ok {
+		// Look for an offset number that is not already in use.
+		// stepping by ResourceSliceImexChannelLimit to avoid overlapping with other cliques.
+		for offset = 0; offset < driverImexChannelLimit; offset += resourceSliceImexChannelLimit {
+			used := false
+			for _, v := range offsets[domain] {
+				if offset == v {
+					used = true
+					break
+				}
+			}
+			if !used {
+				break
+			}
+		}
+
+		// If we reach the limit, return an error.
+		if offset >= driverImexChannelLimit {
+			return -1, fmt.Errorf("error adding IMEX domain %s: channel limit reached", id[0])
+		}
+		offsets[domain][cliqueID] = offset
+	}
```

Comment:
It feels like this should be able to be implemented more simply, but I haven't thought deeply about it yet.

Reply:
Something like:
```
// Track used offsets for the current imexDomain
usedOffsets := make(map[int]struct{})
for _, v := range offsets[domain] {
	usedOffsets[v] = struct{}{}
}

// Look for the first unused offset, stepping by resourceSliceImexChannelLimit
for offset = 0; offset < driverImexChannelLimit; offset += resourceSliceImexChannelLimit {
	if _, exists := usedOffsets[offset]; !exists {
		break
	}
}

// If we reach the limit, return an error
if offset == driverImexChannelLimit {
	return -1, fmt.Errorf("error adding IMEX domain %s: channel limit reached", imexDomain)
}

offsets[imexDomain][cliqueID] = offset
```

Reply:
suggestion accepted